Source/WPF/MyMoney/Controls/TrendGraph.xaml.cs

```diff
@@ -267,45 +267,56 @@ private TrendGraphSeries AddSeries(DateTime start, DateTime end, Color color)
             IList<ChartValue> timeData = s.Values;
             s.BeginUpdate();
 
-            bool first = true;
-            double balance = this.account != null ? (double)this.account.OpeningBalance : 0;
+            bool firstTransaction = true;
+            double balance = 0;
             DateTime last = start;
             Transaction lastt = null;
-            bool started = this.account != null;
 
             foreach (object row in data)
             {
                 Transaction t = row as Transaction;
                 if (t == null) continue;
 
                 if ( t.Account == this.account || // showing transactions for an account
-                    (account == null )) // showing transactions by category // && ((!t.IsBudgeted && t.Account.IsBudgeted))
+                    (this.account == null )) // showing transactions by category // && ((!t.IsBudgeted && t.Account.IsBudgeted))
                 {
-                    decimal v = t.GetCategorizedAmount(this.category);
+                    // If the trend includes the first transaction, also include the opening balance
+                    if (firstTransaction && this.account != null &&  t.Date >= start)
+                    {
+                        balance = (double)this.account.OpeningBalance;

```

Comment:
this is wrong.  The TrendGraph "data" member is all the rows in the Account, so what this loop was doing previously is counting the balance from the very start (starting with OpeningBalance).  Computing the correct balance of a random range of Transactions is hard to do any other way.  With your fixes here I'm finding my checking account shows the wrong balances in the Trend graph, they do not match the balance shown in the transaction view.

Reply:
I misread the loop and thought the intent was to only start accumulating based on the start date.
Accumulating from the beginning is actually easier and makes more sense.
Fixed

Reply:
Ok, the new code looks good, thanks.